packages/material-ui/src/CardMedia/CardMedia.js

```diff
@@ -1,63 +1,93 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import { chainPropTypes } from '@material-ui/utils';
-import withStyles from '../styles/withStyles';
+import { chainPropTypes, deepmerge } from '@material-ui/utils';
 
-export const styles = {
-  /* Styles applied to the root element. */
-  root: {
-    display: 'block',
-    backgroundSize: 'cover',
-    backgroundRepeat: 'no-repeat',
-    backgroundPosition: 'center',
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import useThemeProps from '../styles/useThemeProps';
+import experimentalStyled from '../styles/experimentalStyled';
+import { getCardMediaUtilityClass } from './cardMediaClasses';
+
+const MEDIA_COMPONENTS = ['video', 'audio', 'picture', 'iframe', 'img'];
+const IMAGE_COMPONENTS = ['picture', 'img'];
+
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+
+  const isMediaComponent = MEDIA_COMPONENTS.indexOf(styleProps.component) !== -1;
```

Comment:
How about we add `isMediaComponent` and `isImageComponent` in the `styleProps` and reuse them in all three places?

Reply:
Probably something like https://github.com/mui-org/material-ui/pull/24625/files#r564683087 and we can use `styleProps.isMediaComponent` for example in both the `overridesResolver`, `useUtilityClasses` and the `styled()` funciton